toxcore/onion_announce.c

```diff
@@ -44,8 +52,21 @@ struct Onion_Announce {
     uint8_t secret_bytes[CRYPTO_SYMMETRIC_KEY_SIZE];
 
     Shared_Keys shared_keys_recv;
+
+    uint16_t extra_data_max_size;
+    pack_extra_data_cb *extra_data_callback;
+    void *extra_data_object;
 };
 
+// TODO(Jfreegman): Uncomment this when we merge the rest of new groupchats
+// void onion_announce_extra_data_callback(Onion_Announce *onion_a, uint16_t extra_data_max_size,
```

Comment:
You could keep it in, make it `non_null(1) nullable(2, 3)` and call it from the `new` function with `nullptr`s.

Reply:
Done